### ISSUE TITLE
resource/aws_db_instance: kms_key_id is misleading into key id

### ISSUE THREAD

**Issue by maciej-wilczynski** [hashicorp/terraform-provider-aws#25978]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version

Terraform: v1.2.5
AWS Provider versions: v4.22.0

### Affected Resource(s)

* aws_db_instance

### Terraform Configuration Files

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

```hcl
resource "aws_db_instance" "default" {
  allocated_storage    = 10
  engine               = "mysql"
  engine_version       = "5.7"
  instance_class       = "db.t3.micro"
  name                 = "mydb"
  username             = "foo"
  password             = "foobarbaz"
  parameter_group_name = "default.mysql5.7"
  skip_final_snapshot  = true

  storage_encrypted = true
  kms_key_id        = aws_kms_key.this.id

  performance_insights_enabled    = true
  performance_insights_kms_key_id = aws_kms_key.this.id
}
```

### Expected Behavior

The variable name should point into ARN usage like `kms_key_arn` or `performance_insights_kms_key_arn`

### Actual Behavior

```
Error: "performance_insights_kms_key_id" (xxxxxxx-xxxx-xxxx-xxxxx-xxxxxxxxxx) is an invalid ARN: arn: invalid prefix
with aws_db_instance.default
on main.tf line 16, in resource "aws_db_instance" "default":
  performance_insights_kms_key_id = aws_kms_key.this.id
```

### Steps to Reproduce

1. `terraform plan`

### Important Factoids

The documentation [Argument Reference](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/db_instance#argument-reference) explains that ARN is expected here

**Comment by justinretzolk** [comment]
Hey @maciej-wilczynski 👋 Thank you for taking the time to raise this! Just to make sure I fully understand the ask here, you're essentially looking for the `kms_key_id` and `performance_insights_key_id` arguments to be renamed to `kms_key_arn` and `performance_insights_key_arn` respectively, correct?

**Comment by maciej-wilczynski** [comment]
Yes, it's correct. Using key_id prefix is misleading into ID of the key rather than ARN

**Comment by AdamTylerLynch** [comment]
The [AWS API documentation states](https://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_CreateDBCluster.html#:~:text=Required%3A%20No-,KmsKeyId,-The%20AWS%20KMS) that KmsKeyId is the proper attribute name. Also of note, KmsKeyId accepts a string value that can be an ARN, key ID, alias ARN, or alias name for a KMS key.

The proper change would be to modify the validation function to properly support the same value types the API supports.

**Comment by AdamTylerLynch** [comment]
Additional details on KMS Key ID allowed values:

https://docs.aws.amazon.com/kms/latest/developerguide/concepts.html#key-id
